test/integration/kueuectl/list_test.go

```diff
@@ -194,45 +200,49 @@ very-long-cluster-queue-name            0                   0
-				streams, _, output, errOutput := genericiooptions.NewTestIOStreams()
-				configFlags := CreateConfigFlagsWithRestConfig(cfg, streams)
-				executeTime := time.Now()
-				kueuectl := app.NewKueuectlCmd(app.KueuectlOptions{ConfigFlags: configFlags, IOStreams: streams, Clock: testingclock.NewFakeClock(executeTime)})
-
-				kueuectl.SetArgs([]string{"list", "workload", "--field-selector",
-					fmt.Sprintf("metadata.name=%s", wl1.Name), "--namespace", ns.Name})
-				err := kueuectl.Execute()
-
-				gomega.Expect(err).NotTo(gomega.HaveOccurred(), "%s: %s", err, output)
-				gomega.Expect(errOutput.String()).Should(gomega.BeEmpty())
-				gomega.Expect(output.String()).Should(gomega.Equal(fmt.Sprintf(`NAME   JOB TYPE   JOB NAME   LOCALQUEUE   CLUSTERQUEUE   STATUS    POSITION IN QUEUE   AGE
+				gomega.Eventually(func(g gomega.Gomega) {
+					streams, _, output, errOutput := genericiooptions.NewTestIOStreams()
+					configFlags := CreateConfigFlagsWithRestConfig(cfg, streams)
+					executeTime := time.Now()
+					kueuectl := app.NewKueuectlCmd(app.KueuectlOptions{ConfigFlags: configFlags, IOStreams: streams, Clock: testingclock.NewFakeClock(executeTime)})
+
+					kueuectl.SetArgs([]string{"list", "workload", "--field-selector",
+						fmt.Sprintf("metadata.name=%s", wl1.Name), "--namespace", ns.Name})
+					err := kueuectl.Execute()
+
+					g.Expect(err).NotTo(gomega.HaveOccurred(), "%s: %s", err, output)
+					g.Expect(errOutput.String()).Should(gomega.BeEmpty())
+					g.Expect(output.String()).Should(gomega.Equal(fmt.Sprintf(`NAME   JOB TYPE   JOB NAME   LOCALQUEUE   CLUSTERQUEUE   STATUS    POSITION IN QUEUE   AGE
 wl1                          lq1                         PENDING                       %s
 `,
-					duration.HumanDuration(executeTime.Sub(wl1.CreationTimestamp.Time)))))
+						duration.HumanDuration(executeTime.Sub(wl1.CreationTimestamp.Time)))))
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
 			})
 
 			// Simple client set that are using on unit tests not allow paging.
 			ginkgo.It("Should print workloads list with paging", func() {
-				streams, _, output, errOutput := genericiooptions.NewTestIOStreams()
-				configFlags := CreateConfigFlagsWithRestConfig(cfg, streams)
-				executeTime := time.Now()
-				kueuectl := app.NewKueuectlCmd(app.KueuectlOptions{ConfigFlags: configFlags, IOStreams: streams, Clock: testingclock.NewFakeClock(executeTime)})
-
-				os.Setenv(list.KueuectlListRequestLimitEnvName, "1")
-				kueuectl.SetArgs([]string{"list", "workload", "--namespace", ns.Name})
-				err := kueuectl.Execute()
-
-				gomega.Expect(err).NotTo(gomega.HaveOccurred(), "%s: %s", err, output)
-				gomega.Expect(errOutput.String()).Should(gomega.BeEmpty())
-				gomega.Expect(output.String()).Should(gomega.Equal(fmt.Sprintf(`NAME                      JOB TYPE   JOB NAME   LOCALQUEUE                   CLUSTERQUEUE   STATUS    POSITION IN QUEUE   AGE
+				gomega.Eventually(func(g gomega.Gomega) {
```

Comment:
This test failed here: https://github.com/kubernetes-sigs/kueue/issues/2353#issuecomment-2158668997 due to time shift, which suggests still some parts of the code use real clock. Could you explain why the test failed, then we can think if wrapping with Eventually is the solution we want. Maybe we could make sure all parts of the code use the same fake clock consistently instead.

Reply:
Looks like it just mixed workload creation times.